hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -222,53 +231,203 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
+                    && tableTypeProp.equalsIgnoreCase(
+                        table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP)))
+        .map(table -> TableIdentifier.of(namespace, table.getTableName()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  @SuppressWarnings("FormatStringAnnotation")
+  public void renameView(TableIdentifier from, TableIdentifier originalTo) {
+    if (!namespaceExists(originalTo.namespace())) {
+      throw new NoSuchNamespaceException(
+          "Cannot rename %s to %s. Namespace does not exist: %s",
+          from, originalTo, originalTo.namespace());
+    }
+    renameEntity(from, originalTo, "View");
+  }
 
-    TableIdentifier to = removeCatalogName(originalTo);
+  private void renameEntity(
+      TableIdentifier fromIdentifierEntity, TableIdentifier toIdentifierEntity, String entityType) {
+    if (!isValidIdentifier(fromIdentifierEntity)) {
+      throw new NoSuchViewException("Invalid identifier: %s", fromIdentifierEntity);
+    }
+
+    TableIdentifier to = removeCatalogName(toIdentifierEntity);
     Preconditions.checkArgument(isValidIdentifier(to), "Invalid identifier: %s", to);
 
     String toDatabase = to.namespace().level(0);
-    String fromDatabase = from.namespace().level(0);
-    String fromName = from.name();
+    String fromDatabase = fromIdentifierEntity.namespace().level(0);
+    String fromName = fromIdentifierEntity.name();
 
     try {
-      Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, from));
+      Table fromEntity = clients.run(client -> client.getTable(fromDatabase, fromName));
+      if (entityType.equalsIgnoreCase("Table")) {
+        HiveOperationsBase.validateTableIsIceberg(
+            fromEntity, CatalogUtil.fullTableName(name, fromIdentifierEntity));
+      } else {
+        HiveOperationsBase.validateTableIsIcebergView(
+            fromEntity, CatalogUtil.fullTableName(name, fromIdentifierEntity));
+      }
+
+      validateToTableForRename(fromIdentifierEntity, to);
 
-      table.setDbName(toDatabase);
-      table.setTableName(to.name());
+      fromEntity.setDbName(toDatabase);
+      fromEntity.setTableName(to.name());
 
       clients.run(
           client -> {
-            MetastoreUtil.alterTable(client, fromDatabase, fromName, table);
+            MetastoreUtil.alterTable(client, fromDatabase, fromName, fromEntity);
             return null;
           });
 
-      LOG.info("Renamed table from {}, to {}", from, to);
+      LOG.info("Renamed {} from {}, to {}", entityType, fromIdentifierEntity, to);
 
-    } catch (NoSuchObjectException e) {
-      throw new NoSuchTableException("Table does not exist: %s", from);
+    } catch (NoSuchObjectException | NoSuchViewException e) {
+      if (entityType.equalsIgnoreCase("Table")) {
+        throw new NoSuchTableException("Table does not exist: %s", fromIdentifierEntity);
```

Comment:
why is the error msg for table and view different? I think this should also mention `Cannot rename %s to %s. Table does not exist...`